src/core/polyhedral/memory_promotion_heuristic.cc

```diff
@@ -315,6 +315,48 @@ bool isCoalesced(
   return true;
 }
 
+bool isPromotableToRegisterBelowThreads(
+    const ThreadIdxxScheduleDepthState& threadIdxxScheduleDepthState,
+    const TensorReferenceGroup& group,
+    isl::union_map schedule,
+    size_t nThreads,
+    isl::union_set activePoints) {
+  auto originalAccesses = group.originalAccesses();
+
+  // Return early if more than one element needs to be stored in registers.
+  // TODO: support arrays in registers if they are only accessed with constant
+  // subscripts, e.g. if the inner loops are fully unrolled.
```

Comment:
Quick question regarding unrolling. 
Are you planning to unroll at this point in the mapping or to have some deferred mechanism that also involves codegen, or something else?

Reply:
or maybe you just meant full tiles here?

Reply:
I meant situations like accessing 
```c
for (int i = 0; i < 2; ++i) {
  A[2*threadIdx.x + i];
```
that would require having `register_A[2]` accessed as `register_A[i]`, which would go to local memory instead of registers.  Unless the `i` loop is unrolled, there is not point in promoting this access.

Because autotuner-controlled unrolling happens earlier in the transformation pipeline, I was planning to just if the loop is requested for unroll.